### ISSUE TITLE
[Atomic2D] Support navigation and 2D physics on the same orientation plane

### ISSUE THREAD

**Issue by Alan-FGR** [AtomicGameEngine/AtomicGameEngine#1636]
Navigation works on the XZ plane, and 2D physics works on the XY plane, thus making them fully unusable together, what could be useful for planar 2D top-down games like shooters, racers or RTSs where you don't need to use the 3D physics.

**Comment by mattbenic** [comment]
I haven't touched navigation at all. Is it implemented in a third party lib, or as part of Atomic core? If the latter, is it something inherited from Urho?

**Comment by Alan-FGR** [comment]
Navigation is Recast/Detour, but it doesn't require specific axiality, now that I've given it a second thought though, navigation uses mesh for generation and we may have to provide a better approach for that when it comes to 2d since most games are sprite based so you can't pass that data for navmesh generation.

**Comment by mattbenic** [comment]
So this is more of a general "provide navigation for 2D" task. We discussed dropping maintenance of 2D completely previously, that would negate this completely.

**Comment by Alan-FGR** [comment]
Indeed. Marking as potentially irrelevant since if we're dropping 2d it is indeed.
